### PR TITLE
Fix ArgumentOutOfRangeException at MatchParamNamesWithInstantiatedFromPass

### PR DESCRIPTION
**Pull request by AhmedZero**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/mono/CppSharp?pullRequest=1641) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/mono/CppSharp?pullRequest=1641) before we can accept your contribution.<br/><br/>:x: AhmedZero [sign now](https://cla.dotnetfoundation.org/mono/CppSharp?pullRequest=1641)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/mono/CppSharp?pullRequest=1641) it.</sub>

File: tests/CSharp/CSharpTemplates.h [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 #include <string>
 #include <map>
-
+#include <array>
```
Comment (ddobrev) [inline tests/CSharp/CSharpTemplates.h:8]: Please keep the new line between #include-s and the code itself

File: tests/CSharp/CSharpTemplates.h [hunk 2]
```diff
@@ -842,3 +842,17 @@ class ClassWithNonTypeTemplateArgument
 
 class SpecializationOfClassWithNonTypeTemplateArgument : public ClassWithNonTypeTemplateArgument<0>
 { };
+template<std::size_t N>
+class DLL_API FloatArrayF
+{
+protected:
+    std::array< double, N > values;
```
Comment (ddobrev) [inline tests/CSharp/CSharpTemplates.h:849]: Please decrease the test case as much as possible until it's just enough to catch the bug. Just one example: if this `std::array< double, N > values;` isn't necessary for reproduction, remove it. Please remember that when you're done, you need to temporarily disable your fix to ensure the test still exposes the bug.

**Review by ddobrev — CHANGES_REQUESTED** [review]

**Comment by AhmedZero** [comment]
delete all unnecessary lines.

**Pull request merged**